coral/src/app/features/topics/request/TopicEditRequest.test.tsx

```diff
@@ -0,0 +1,290 @@
+import { Context as AquariumContext } from "@aivenio/aquarium";
```

Comment:
The test is missing to confirm some cases:

- confirm it calls `getEnvironmentsForTopicRequest` on render
- confirm it calls `getTopicDetailsPerEnv` with the right arguments (with env from search params or empty string)
- confirms for both api calls that errors or empty responses are handled and show correctly
- user can submit data and sees error if error happens

(cases that are covered)
- shows expected data when all data is fetched and correct
- user sees correct validation errors for invalid input and can't submit
- user can submit form successfully. (covered)
- user can cancel form 
-> without having changed input
-> with having changed input

Reply:
> confirm it calls getEnvironmentsForTopicRequest on render

This is tested by the fact that `editTopic` is called with the `environment` value from `mockGetAllEnvironmentsForTopicAndAcl` returned value, isn't it? Isn't testing the outcome better than testing implementation detail? We may get that environment value from elsewhere in the future. Probably should be made more explicit though, by passing `mockEnvironments[0],` instead of inlining the value:

```
      expect(mockEditTopic).toHaveBeenCalledWith({
        advancedConfiguration: '{"cleanup.policy":"compact"}',
        description: "Topic description",
        environment: mockEnvironments[0],
        remarks: "please approve",
        replicationfactor: "1",
        topicname: "test-topic-name",
        topicpartitions: "2",
      });
```

> confirm it calls getTopicDetailsPerEnv with the right arguments (with env from search params or empty string)
Same as above, could look like this?


```
      expect(mockEditTopic).toHaveBeenCalledWith({
        advancedConfiguration: JSON.stringify(
          mockTopicDetails.topicContents.advancedTopicConfiguration
        ),
        description: mockTopicDetails.topicContents.description,
        environment: mockEnvironments[0],
        remarks: "please approve",
        replicationfactor: mockTopicDetails.topicContents.noOfReplicas,
        topicname: TOPIC_NAME,
        topicpartitions: String(mockTopicDetails.topicContents.noOfPartitions),
      });
```

The error handling should be tested, though, yes :o I will test it if the implementation here seems correct? https://github.com/aiven/klaw/pull/1436#discussion_r1257798804

Reply:
Implementing what I said in previous comment: [bf94752](https://github.com/aiven/klaw/pull/1436/commits/bf94752d3365380c03e6764851ff0b36c02bc728)

Reply:
And testing: [76bdb25](https://github.com/aiven/klaw/pull/1436/commits/76bdb255b3bcfe7f60c3fe2bbd0c8828b618b00f)